EESSI-pilot-2020.09.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+#
+# Script to install EESSI pilot software stack (version 2020.09)
+#
+
+TOPDIR=$(dirname $(realpath $0))
+
+function echo_green() {
+    echo -e "\e[32m$1\e[0m"
+}
+
+function echo_red() {
+    echo -e "\e[31m$1\e[0m"
+}
+
+function echo_yellow() {
+    echo -e "\e[33m$1\e[0m"
+}
+
+function error() {
+    echo_red "ERROR: $1" >&2
+    exit 1
+}
+
+TMPDIR=$(mktemp -d)
+
+echo ">> Setting up environment..."
+export CVMFS_REPO="/cvmfs/pilot.eessi-hpc.org"
+export EESSI_PILOT_VERSION="2020.09"
+export ARCH=$(uname -m)
+export EESSI_PREFIX=${CVMFS_REPO}/${EESSI_PILOT_VERSION}
+export EPREFIX=${EESSI_PREFIX}/compat/${ARCH}
+
+# make sure that $PATH starts with $CVMFS_REPO
+# (if not, we're not running in the environment set up by 'startprefix')
+if [[ $PATH = ${CVMFS_REPO}* ]]; then
+    echo_green ">> It looks like we're in a Gentoo Prefix environment, good!"
+else
+    error "Not running in Gentoo Prefix environment, run '${EPREFIX}/startprefix' first!"
+fi
+
+echo ">> Initializing Lmod..."
+source $EPREFIX/usr/lmod/*/init/bash
+ml_version_out=$TMPDIR/ml.out
+ml --version &> $ml_version_out
+if [[ $? -eq 0 ]]; then
+    echo_green ">> Found Lmod ${LMOD_VERSION}"
+else
+    error "Failed to initialize Lmod?! (see output in ${ml_version_out}"
+fi
+
+echo ">> Determining software subdirectory to use for current build host..."
+export EESSI_SOFTWARE_SUBDIR=$(python3 $TOPDIR/eessi_software_subdir.py)
+if [[ -z ${EESSI_SOFTWARE_SUBDIR} ]]; then
+    error "Failed to determine software subdirectory?!"
+else
+    echo_green ">> Using ${EESSI_SOFTWARE_SUBDIR} as software subdirectory!"
+fi
+
+echo ">> Configuring EasyBuild..."
+export EASYBUILD_PREFIX=/tmp/${USER}/easybuild
+export EASYBUILD_INSTALLPATH=${EESSI_PREFIX}/software/${EESSI_SOFTWARE_SUBDIR}
+export EASYBUILD_SOURCEPATH=/tmp/$USER/easybuild/sources:${EESSI_SOURCEPATH}
+
+# just ignore OS dependencies for now, see https://github.com/easybuilders/easybuild-framework/issues/3430
+export EASYBUILD_IGNORE_OSDEPS=1
+
+export EASYBUILD_SYSROOT=${EPREFIX}
+
+export EASYBUILD_DEBUG=1
+export EASYBUILD_TRACE=1
+
+export EASYBUILD_RPATH=1
+export EASYBUILD_FILTER_ENV_VARS=LD_LIBRARY_PATH
+
+export EASYBUILD_FILTER_DEPS=Autoconf,Automake,Autotools,binutils,bzip2,gettext,libreadline,libtool,M4,ncurses,XZ,zlib
```

Comment:
More stuff we should consider adding to `--filter-deps`:

* `Bison`
* `cURL`
* `expat`
* `flex`
* `help2man`
* `pkg-config`

Reply:
Do we still need to keep/discuss this somewhere (in an issue)?

Reply:
Yes, we should have a discussion on what we (won't) include in this list for the next iteration (2020.10).

We've hit several issues by leveraging the `binutils`, `zlib`, `ncurses` provided by Prefix, for different reasons.
That's not a reason to not filter those, but it does show that relying on stuff from the compat layer could have an impact (changes required in easyblocks/easyconfigs to make it work).

Reply:
see #23